File: leetcode/dynamicPlanning/2110.cpp
#include <algorithm>
#include <functional> //包含std::greater 和 std::less
#include <iostream>
#include <limits.h>
#include <map>
#include <math.h>
#include <numeric>
#include <queue>
#include <set>
#include <stack>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right)
      : val(x), left(left), right(right) {}
};
struct ListNode {
  int val;
  ListNode *next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
  long long getDescentPeriods(vector<int> &prices) {
    int n = prices.size();
    int last_day = 1;
    long long res = last_day;
    for (int i = 1; i < n; i++) {
      if (prices[i] == prices[i - 1] - 1) {
        last_day += 1;
      }else{
        last_day = 1;
      }
      res += last_day;
    }
    return res;
  }
};
int main() {
  // 示例二叉树
  Solution solution;
  vector<int> nums = {7, 1, 5, 3, 6, 4};
  int result = solution.getDescentPeriods(nums);
  std::cout << "result: " << result << std::endl;
}

File: leetcode/dynamicPlanning/2110.py
from typing import List
from queue import Queue


class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        n: int = len(prices)
        dp: int = 1
        res: int = 1
        for i in range(1, n):
            if prices[i] == prices[i - 1] - 1:
                dp += 1
            else:
                dp = 1
            res += dp
        return res


if __name__ == "__main__":
    solution = Solution()
    grid = [
        ["1", "1", "0", "0", "0"],
        ["1", "1", "0", "0", "0"],
        ["0", "0", "1", "0", "0"],
        ["0", "0", "0", "1", "1"],
    ]
    nums = [1, 2, 3]
    result = solution.getDescentPeriods(nums)
    print(f"result = {result}")
